src/mscorlib/src/System/Lazy.cs

```diff
@@ -254,7 +456,7 @@ internal T ValueForDebugDisplay
                 {
                     return default(T);
                 }
-                return ((Boxed)m_boxed).m_value;
+                return Value;
```

Comment:
I changed `ValudForDebugDisplay`; do we still need the `NotifyOfCrossThreadDependency`?